### DISCUSSION TITLE
How to show raw critic text?

### DISCUSSION THREAD

**Discussion by zillionare** [facelessuser/pymdown-extensions#1927] [Q&A] [answered]
In my document, I need to show both original critic markup and its rendered output to explain to readers that how to use critic function. Please someone tell me how to do this?

in official document, it says:

```
=== "Output"
    ```md-render
    ---
    extensions:
    - pymdownx.critic
    extension_configs:
      pymdownx.critic:
        mode: accept
    ---
    Here is some {--*incorrect*--} Markdown.  I am adding this{++ here++}.  Here is some more {--text
    that I am removing--}text.  And here is even more {++text that I
    am ++}adding.{~~

    ~>  ~~}Paragraph was deleted and replaced with some spaces.{~~  ~>

    ~~}Spaces were removed and a paragraph was added.

    And here is a comment on {==some
     text==}{>>This works quite well. I just wanted to comment on it.<<}. Substitutions {~~is~>are~~} great!

    General block handling.

    {--

    * test remove
    * test remove
    * test remove
        * test remove
    * test remove

    --}

    {++

    * test add
    * test add
    * test add
        * test add
    * test add

    ++}
    ```

=== "Markdown"
    ```critic-markup
    Here is some {--*incorrect*--} Markdown.  I am adding this{++ here++}.  Here is some more {--text
     that I am removing--}text.  And here is even more {++text that I 
     am ++}adding.{~~

    ~>  ~~}Paragraph was deleted and replaced with some spaces.{~~  ~>

    ~~}Spaces were removed and a paragraph was added.

    And here is a comment on {==some
     text==}{>>This works quite well. I just wanted to comment on it.<<}. Substitutions {~~is~>are~~} great!

    General block handling.

    {--

    * test remove
    * test remove
    * test remove
        * test remove
    * test remove

    --}

    {++

    * test add
    * test add
    * test add
        * test add
    * test add

    ++}
    ```
```

have no idea what md-render and critic-markup do here.

**Comment by facelessuser** [comment, marked as answer]
`crtic-markup` is a custom Pygments lexer that I created for my documents: https://github.com/facelessuser/pymdown-lexers. It's not really documented as I had intended to only use it for myself. There are no actual releases for it as I just include it in my requirements like this https://github.com/facelessuser/pymdown-extensions/blob/main/requirements/docs.txt#L2.

`md-render` is also a custom plugin I use for my documents. It is also not documented as it was just a solution for my docs. I wanted to not apply Critic Markup to my documents, but instead, only activate the usage of it in sandboxes. So I created custom fenced code blocks where I can enable certain markdown plugins only in those code blocks. It works well for certain plugins at least.

Custom plugin: https://github.com/facelessuser/pymdown-extensions/blob/main/tools/pymdownx_md_render.py
How I configure it: https://github.com/facelessuser/pymdown-extensions/blob/main/mkdocs.yml#L103

Feel free to use it at your own risk.

**Comment by zillionare** [comment]
thank @facelessuser
